internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -428,6 +440,74 @@ func (r *ClusterDeploymentsReconciler) updateInstallConfigOverrides(ctx context.
 	return nil
 }
 
+func (r *ClusterDeploymentsReconciler) syncManifests(ctx context.Context, cluster *common.Cluster, manifests map[string]string, alreadyCreatedManifests models.ListManifests) error {
+	r.Log.Debugf("Going to sync list of given with already created manifests")
+
+	for _, manifest := range alreadyCreatedManifests {
+		if _, ok := manifests[manifest.FileName]; !ok {
+			r.Log.Infof("Deleting cluster %s manifest %s", cluster.ID, manifest.FileName)
+			_ = r.Manifests.DeleteClusterManifestInternal(ctx, operations.DeleteClusterManifestParams{
+				ClusterID: *cluster.ID,
+				FileName:  manifest.FileName,
+				Folder:    swag.String(models.ManifestFolderOpenshift),
+			})
+		} else {
+			delete(manifests, manifest.FileName)
+		}
+	}
+
+	for filename, manifest := range manifests {
+		r.Log.Infof("Creating cluster %s manifest %s", cluster.ID, filename)
+		_, err := r.Manifests.CreateClusterManifestInternal(ctx, operations.CreateClusterManifestParams{
+			ClusterID: *cluster.ID,
+			CreateManifestParams: &models.CreateManifestParams{
+				Content:  swag.String(base64.StdEncoding.EncodeToString([]byte(manifest))),
+				FileName: swag.String(filename),
+				Folder:   swag.String(models.ManifestFolderOpenshift),
+			}})
+		if err != nil {
+			r.Log.WithError(err).Errorf("Failed to create cluster %s manifest %s", cluster.ID, filename)
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *ClusterDeploymentsReconciler) addCustomManifests(ctx context.Context, clusterDeployment *hivev1.ClusterDeployment,
+	cluster *common.Cluster) error {
+
+	if !common.IsInPreInstallationState(cluster) {
```

Comment:
Instead of skipping consider validating that the `alreadyCreatedManifests` are synced with the `ManifestsConfigMapRef`.
If not set a condition to "Failed" with a relevant error.

Reply:
I don't want to do anything on manifests if we are already installing. We are adding more manifests on installation process and i don't want to sync in that case. In any case it will be too late.